index.js

```diff
@@ -91,8 +90,7 @@ function equal(a, b) {
 
     // custom handling for React
     for (i = length; i-- !== 0;) {
```

Comment:
In my change I've initialised `i = -1` which allows us to replace all of these for-loops with just:

```suggestion
    while (++i < length) {
```

That does change the iteration order here to forwards instead of backwards, but shouldn't influence perf.

This is the same for these lines:

- https://github.com/FormidableLabs/react-fast-compare/blob/5a9c1ac30964a49c38d31a4c093299f677930c53/index.js#L21
- https://github.com/FormidableLabs/react-fast-compare/blob/5a9c1ac30964a49c38d31a4c093299f677930c53/index.js#L70
- https://github.com/FormidableLabs/react-fast-compare/blob/5a9c1ac30964a49c38d31a4c093299f677930c53/index.js#L83

Reply:
Tried this one out. Because we have the constraints of needing to do independent loops to avoid recursion, I've had  to add `i = -1` before all but the first one and size increases 6 bytes min+gz.